internals-js/src/definitions.ts

```diff
@@ -3062,7 +3062,11 @@ export class Directive<
     assert(definition, () => `Cannot include default values for arguments: cannot find directive definition for ${this.name}`);
     const updated = Object.create(null);
     for (const argDef of definition.arguments()) {
-      updated[argDef.name] = withDefaultValues(this._args[argDef.name], argDef);
+      const argValue = withDefaultValues(this._args[argDef.name], argDef);
+      // Note that argValue could be '0' or something falsy here, so we must explicitly check === undefined
+      if (argValue === undefined) {
```

Comment:
I believe you meant `!==` here (I believe that's at least one of the reason for test breakage).